.github/workflows/main.yml

```diff
@@ -155,6 +165,31 @@ jobs:
         env:
           CC: ${{ matrix.CC }}
           CXX: ${{ matrix.CXX }}
+      - name: Install llvm tools (for llvm-ar)
+        if: startsWith(matrix.build, 'cross-macos') || startsWith(matrix.build, 'cross-ios')
+        run: sudo apt-get install llvm
+      - name: Download macOS SDK
+        if: startsWith(matrix.build, 'cross-macos')
+        run: |
+          wget https://github.com/phracker/MacOSX-SDKs/releases/download/11.3/MacOSX11.3.sdk.tar.xz
+          tar -xf MacOSX11.3.sdk.tar.xz
+          echo "SDKROOT=$(pwd)/MacOSX11.3.sdk" >> $GITHUB_ENV
+      - name: Download iOS SDK
+        if: startsWith(matrix.build, 'cross-ios')
+        run: |
+          wget https://github.com/xybp888/iOS-SDKs/releases/download/iOS18.1-SDKs/iPhoneOS18.1.sdk.zip
+          unzip iPhoneOS18.1.sdk.zip
+          echo "SDKROOT=$(pwd)/iPhoneOS18.1.sdk" >> $GITHUB_ENV
+      - name: Set up Apple cross-compilation
+        if: startsWith(matrix.build, 'cross-macos') || startsWith(matrix.build, 'cross-ios')
+        run: |
+          # Test with clang/llvm for now, has better cross-compilation support (GCC requires downloading a different toolchain)
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
+          echo "AR=llvm-ar" >> $GITHUB_ENV
```

Comment:
It is unclear to me why `llvm-ar` is required here, it seems to not be on my local Ubuntu 24 VM? But if we don't include it, we fail with:
```console
rust-lld: error: /home/runner/work/cc-rs/cc-rs/target/aarch64-apple-darwin/debug/build/cc-test-5875314a6fda6e38/out/libOptLinkage.a: archive has no index; run ranlib to add one
rust-lld: error: undefined symbol: _answer
```

Reply:
No idea maybe there's some apple/clang stuff default ar not supporting?